wp-content/plugins/goonj-blocks/build/functions.php

```diff
@@ -15,4 +15,203 @@ function gb_format_time_range($start_date, $end_date, $format = 'h:i A') {
         throw new InvalidArgumentException('End time must be after start time');
     }
     return $start_date->format($format) . ' - ' . $end_date->format($format);
-}
\ No newline at end of file
+}
+/**
+ * Generate induction slots based on contact's location, registration date, and induction activity details.
+ * 
+ * @param int|null $contactId The ID of the contact for whom to generate slots.
+ * @param int $days The number of days to look ahead for slot generation (default is 30 days).
+ * @return array An array of induction slots or empty if criteria aren't met.
+ */
+function generate_induction_slots($contactId = null, $days = 30) {
+    // Validate contactId
+    if (empty($contactId) || !is_numeric($contactId)) {
+        \Civi::log()->warning('Invalid contactId', ['contactId' => $contactId]);
```

Comment:
As we are using a warning here are we showing the message to the user? If not then why we are not using Debug

Reply:
updated and used info()